### DISCUSSION TITLE
Improve Avalonia WASM Startup Time? Server Side Rendering?

### DISCUSSION THREAD

**Discussion by mw2gd** [AvaloniaUI/Avalonia#15415] [Q&A]
Hello. I'm familiar with Avalonia Desktop development but not with WASM.

From my understanding, the .NET runtime has to be provided to the user as WASM bytecode, which makes startup times pretty slow on the initial loading of the site. 

From looking at documentation on Blazor, it seems they have different Render modes, including an Interactive server-side rendering mode, which gets around the need for WASM, since interaction is handled on the server side. This greatly reduces the initial startup time.

Is there any way to setup an _Avalonia_ project with server side rendering? Is that even possible?

If not, is there any 'magic' that can be done to get around the slow initial startup beside trying to reduce/trim the size of the runtime and app? I realize this is a difficult problem to solve.

**Comment by maxkatz6** [comment]
There is a fundamental difference between how Blazor and Avalonia work. 

Blazor builds a HTML components tree, which can be done on the server side. Or in a mixed mode. Browser will only need to display this logic.

Avalonia is a canvas rendered framework, in a context of browser. You theoretically could render bitmaps on the server and transfer them to the browser, but practically it won't work well.

>  beside trying to reduce/trim the size of the runtime and app

Exactly this. Reduce size, work around trimming parameters. 
Though, keep in mind "small size" != "fast startup time". You can greatly improve startup time by using Mono AOT (in the future, NativeAOT as well, which has many perf benefits). But AOT usually has a bigger app size.
